analyzers/src/SonarAnalyzer.CSharp/Rules/MessageTemplates/LoggingTemplatePlaceHoldersShouldBeInOrder.cs

```diff
@@ -92,12 +94,52 @@ private static SyntaxNode OutOfOrderPlaceholderValue(MessageTemplatesParser.Plac
         return null;
     }
 
-    private static bool? MatchesName(string placeholderName, SyntaxNode placeholderValue) =>
+    private static bool? MatchesName(string placeholderName, SyntaxNode placeholderValue, bool isStrict) =>
         placeholderValue switch
         {
-            MemberAccessExpressionSyntax memberAccess => MatchesName(placeholderName, memberAccess.Name),
-            CastExpressionSyntax cast => MatchesName(placeholderName, cast.Expression),
-            NameSyntax name => new string(placeholderName.Where(char.IsLetterOrDigit).ToArray()).Equals(name.GetName(), StringComparison.OrdinalIgnoreCase),
+            MemberAccessExpressionSyntax memberAccess => MatchesName(placeholderName, memberAccess.Name, isStrict),
+            CastExpressionSyntax cast => MatchesName(placeholderName, cast.Expression, isStrict),
+            NameSyntax name => SimpleStringMatches(placeholderName, name.GetName(), isStrict),
             _ => null
         };
+
+    private static bool SimpleStringMatches(string placeholderName, string argumentName, bool isStrict)
+    {
+        if (isStrict)
+        {
+            return OnlyLetters(placeholderName).Equals(OnlyLetters(argumentName), StringComparison.OrdinalIgnoreCase);
+        }
+        else
+        {
+            var placeholderComponents = SplitByCamelCase(placeholderName);
+            var argumentComponents = SplitByCamelCase(argumentName);
+            return Array.Exists(placeholderComponents, x => Array.Exists(argumentComponents, y => x.Equals(y, StringComparison.OrdinalIgnoreCase)));
```

Comment:
Isn't this the same with:
```suggestion
            return placeholderComponents.Intersect(argumentComponents, StringComparer.OrdinalIgnoreCase).Any();
```